test/specs/init.js

```diff
@@ -29,6 +29,21 @@ module.exports = function (promise) {
       expect(fsm.current).to.be.equal('green');
     });
 
+    it('should accept an Object for events property', function () {
+      StateMachine.Promise = promise;
+
+      var fsm = StateMachine({
+        events: {
```

Comment:
> I have some state machines where this approach will not work as they have multiple events having the same name but starting from different states.

I'd like to understand your use case better.  Why do they have the same name?  If event `foo` can be called in states `a` or `b`, does this event cause transition to state `c` in both cases?  If so, you could specify multiple "from" states in the event definition, using the object notation:

```js
{
  name: 'foo',
  from: ['a', 'b'],
  to: 'c'
}
```

But if it *doesn't* always transition to `c` (or have a conditional), I'm not sure why it has to be the same name?